src/target/llvm/llvm_instance.cc

```diff
@@ -845,10 +847,11 @@ const Array<String> LLVMTargetInfo::GetAllLLVMCpuFeatures() const {
 #else
       MCInfo->getAllProcessorFeatures();
 #endif
-  Array<String> cpu_features;
+  // TVM doesn't have an FFI friendly Set, so use a Map instead for now
+  Map<String, String> cpu_features;
```

Comment:
This looks ugly in python backend, would be std::set better ?

See:
```
$ cat ./tvm-target.py 
#!/usr/bin/python3

target = tvm.target.Target("llvm -mcpu=x86-64")
print( tvm.target.codegen.llvm_get_cpu_features(target) )
```
* BEFORE
```
$ ./tvm-target.py 
["64bit", "64bit-mode", "cmov", "cx8", "fxsr", "idivq-to-divl", "macrofusion", 
"mmx", "nopl", "slow-3ops-lea", "slow-incdec", "sse", "sse2", "vzeroupper", "x87"]
```
* AFTER
```
$ ./tvm-target.py 
{"mmx": "", "cmov": "", "idivq-to-divl": "", "slow-incdec": "", "macrofusion": "", 
"fxsr": "", "slow-3ops-lea": "", "x87": "", "nopl": "", "sse2": "", "vzeroupper": "", 
"64bit": "", "sse": "", "cx8": "", "64bit-mode": ""}

```

Reply:
Good point thanks, I'd forgotten to check this. Due to the restriction on the FFI, perhaps we can fix this on the python side? For example:
```
feature_map = _ffi_api.llvm_get_cpu_features(target)
return set(feature_map.keys())
```
We could also use std::set for the c++ API if desired, we'd just have to make the conversion to/from Map before/after crossing the FFI boundary.

Reply:
> Good point thanks, 

However I think my request concerns the "pedantic" side and ignores the "performance" gains.

>I'd forgotten to check this. Due to the restriction on the FFI, perhaps we can fix this on the python side? For example:
> 
> ```
> feature_map = _ffi_api.llvm_get_cpu_features(target)
> return set(feature_map.keys())
> ```

Super ! 

That would be enough for python side.

> 
> We could also use std::set for the c++ API if desired, we'd just have to make the conversion to/from Map before/after crossing the FFI boundary.

See, well I think ```Map``` is fine and yes is may be the fastest option concerning lookup speeds in hot-paths.

Thanks @lhutton1 !